fiona/ogrext.pyx

```diff
@@ -775,7 +775,11 @@ cdef class Session:
             raise ValueError("Null layer")
 
         result = OGR_L_GetExtent(self.cogr_layer, &extent, 1)
-        return (extent.MinX, extent.MinY, extent.MaxX, extent.MaxY)
+        
+        if result == OGRERR_NONE:
+            return (extent.MinX, extent.MinY, extent.MaxX, extent.MaxY)
+        else:
+            return None
```

Comment:
I'm not sure if this is the right thing to do, as for a user a None bound is probably not expected. Raising an exception would require the modifications of some tests.